erdpy_network/transaction_completion_strategy.py

```diff
@@ -16,6 +27,36 @@ def is_completed(self, transaction: ITransactionOnNetwork) -> bool:
         return not transaction.get_status().is_pending()
 
 
+# this class is similar to the one in erdjs-network-providers
 class TransactionCompletionStrategyOnProxy:
-    def is_completed(self, transaction: ITransactionOnNetwork):
-        raise NotImplementedError('Method is not implemented yet')
+    def is_completed(self, transaction: ITransactionOnNetwork) -> bool:
+        if transaction.get_status().is_pending():
+            return False
+
+        for event in KNOWN_COMPLETION_EVENTS:
+            if transaction.logs.find_first_or_none_event(event):
+                return True
+
+        if self.__is_certainly_move_balance(transaction.data):
+            return transaction.status.is_executed()
+
+        #  Imprecise condition, uncertain completion (usually sufficient, though).
+        #  This is WRONG when (at least): timeOf(block with execution at destination is notarized) < timeOf(the "completedTxEvent" occurs).
+        if transaction.hyperblock_nonce > 0:
+            return True
+
+        return False
+
+    def __is_certainly_move_balance(self, transaction_data: str) -> bool:
+        parts = transaction_data.split("@")
+        prefix = parts[0]
+        other_parts = parts[1:]
+        empty_prefix = not prefix
+        some_parts_are_not_valid_args = not all(
+            map(self.__is_valid_argument, other_parts)
+        )
+
+        return empty_prefix or some_parts_are_not_valid_args
+
+    def __is_valid_argument(self, arg: str) -> bool:
+        return is_padded_hex(arg)
```

Comment:
is this enough? what if the arguments are even lengthed, but they include non-hex chars. will the transaction be treated as move balance?

Reply:
Perhaps we should add a better implementation of `__is_valid_argument`, indeed.

Perhaps we should also adjust the implementation in erdjs:
https://github.com/ElrondNetwork/elrond-sdk-erdjs-network-providers/blob/main/src/transactionCompletionStrategy.ts#L60

However, we can expect (allow) this implementation to fail in case of exotic, badly formatted-transactions (and handle completeness somehow **at the Proxy level** at some point in the future).